src/field/pokemon-move.ts

```diff
@@ -30,14 +34,47 @@ export class PokemonMove {
    */
   public maxPpOverride?: number;
 
-  constructor(moveId: MoveId, ppUsed: number = 0, ppUp: number = 0, virtual: boolean = false, maxPpOverride?: number) {
+  constructor(
+    moveId: MoveId,
+    {
+      pokemonId = 0,
+      ppUsed = 0,
+      ppUp = 0,
+      virtual = false,
+      maxPpOverride,
+    }: {
+      pokemonId?: number;
+      ppUsed?: number;
+      ppUp?: number;
+      virtual?: boolean;
+      maxPpOverride?: number;
+    } = {},
+  ) {
+    this.pokemonId = pokemonId;
     this.moveId = moveId;
     this.ppUsed = ppUsed;
     this.ppUp = ppUp;
     this.virtual = virtual;
     this.maxPpOverride = maxPpOverride;
   }
 
+  private get pokemon(): Pokemon | undefined {
+    return globalScene.getPokemonById(this.pokemonId) ?? undefined;
+  }
+
+  public get name(): string {
+    if (isNil(this.pokemon)) {
+      return this.getMove().name;
+    }
+    const gMaxDescriptor =
+      this.getMove().checkFlag(MoveFlags.G_MAX_MOVE, this.pokemon) && this.pokemon.isMax()
+        ? i18next.t("move:gMaxDescriptor")
+        : "";
+
+    const moveName = this.getMove().name;
+    return i18next.t("move:moveFormatWithGMax", { gMaxDescriptor, moveName });
```

Comment:
This does not have the intended result. `moveFormatWithGMax` puts the space in even if there is no gMaxDescriptor. Maybe there's a way to make a conditional with i18next, dunno.
Otherwise we either need to trim the string before returning it, or:

```suggestion
    const moveName = this.getMove().name;
    const isMax = this.getMove().checkFlag(MoveFlags.G_MAX_MOVE, this.pokemon) && this.pokemon.isMax();
    if (isMax) {
      const gMaxDescriptor = i18next.t("move:gMaxDescriptor");
      return i18next.t("move:moveFormatWithGMax", { gMaxDescriptor, moveName });
    }
    return moveName;
```

This also fixes the UI tests you disabled. They were failing because they look for a TextObject containing "Dragon Claw", but it was " Dragon Claw" because of the space issue (the UI test would still fail on this branch until the locale PR is merged in)

Reply:
Thinking a bit more about this. If we go this route, we could also simply merge `gMaxDescriptor` and `moveFormatWithGMax` in a single key

Reply:
I figure the easiest fix would be to update the keys themselves so that `gMaxDescriptor` includes the space (in English at least)?

Reply:
Yes but that would mean it that key can only ever be used in this very context, and is likely to confuse translators as well. And if it's limited to a single context it might as well be all in a single key I think